src/main/java/org/opensearch/security/dlic/rest/support/Utils.java

```diff
@@ -213,4 +217,28 @@ public static String hash(final char[] clearTextPassword) {
                 .map(field -> prefix + field)
                 .collect(ImmutableSet.toImmutableSet());
     }
+
+    /**
+     * Add prefixes(_opendistro... and _plugins...) to rest API routes
+     * @param routes routes
+     * @return new list of API routes prefixed with _opendistro... and _plugins...
+     *Total number of routes is expanded as twice as the number of routes passed in
+     */
+    public static List<Route> addRoutesPrefix(List<Route> routes){
+        return addRoutesPrefix(routes, "/_opendistro/_security/api", "/_plugins/_security/api");
+    }
+
+    /**
+     * Add customized prefix(_opendistro... and _plugins...)to API rest routes
+     * @param routes routes
+     * @param prefixes all api prefix
+     * @return new list of API routes prefixed with the strings listed in prefixes
+     * Total number of routes will be expanded len(prefixes) as much comparing to the list passed in
+     */
+    public static List<Route> addRoutesPrefix(List<Route> routes, final String... prefixes){
+        return routes.stream().flatMap(
+                r -> Arrays.stream(prefixes)
+                           .map(p -> new Route(r.getMethod(), p + r.getPath())))
+                              .collect(ImmutableList.toImmutableList());
```

Comment:
this code is hard to read without counting `)` as it is impossible to say whether `collect()` is called on the result of `flatMap()` or `map()`.